### PR TITLE
Add an example for batch normalization.

### PR DESCRIPTION
**Pull request by dwf**
Paired with mila-udem/blocks#941.

Attn: @rizar

### CODE REVIEW

**Comment by dwf** [comment]
@cooijmanstim Mind looking at this too?

File: batch_normalization/__init__.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    features = tensor.matrix('features')
+    label = tensor.lvector('label')
+    prediction = mlp.apply(features)
+    cost = CategoricalCrossEntropy().apply(label, prediction)
+    misclass = MisclassificationRate().apply(label, prediction)
+    misclass.name = 'misclass'  # The default name for this is annoyingly long
+    original_cg = ComputationGraph([cost, misclass])
+
+    if batch_normalized:
+        cg = apply_batch_normalization(original_cg)
```
Comment (vdumoulin) [inline batch_normalization/__init__.py]: I'd explain what `apply_batch_normalization` does, i.e. take an inference graph and return a graph where population statistics have been replaced by batch statistics.

File: batch_normalization/__init__.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        # Add updates for population parameters
```
Comment (vdumoulin) [inline batch_normalization/__init__.py]: Maybe add that these updates correspond to maintaining an exponential moving average of the batch statistics?

**Comment by vdumoulin** [comment]
Other than my two inline comments, LGTM.

**Comment by cooijmanstim** [comment]
Sorry, I forgot about this PR. I agree with @vdumoulin and have nothing else to add.

**Comment by rizar** [comment]
Home come it had not been merged?! :)
